capellambse/decl.py

```diff
@@ -197,7 +197,17 @@ def _operate_modify(
             except _UnresolvablePromise as p:
                 yield p.args[0], {"parent": parent, "modify": {attr: value}}
                 continue
-        setattr(parent, attr, value)
+
+        if isinstance(value, list):
+            delattr(parent, attr)
+            yield from _create_complex_objects(promises, parent, attr, value)
+        else:
+            if isinstance(value, dict):
+                obj = getattr(parent, attr)
+                for k, v in value.items():
+                    setattr(obj, k, v)
+            else:
+                setattr(parent, attr, value)
```

Comment:
```suggestion
        elif isinstance(value, dict):
            obj = getattr(parent, attr)
            for k, v in value.items():
                setattr(obj, k, v)
        else:
            setattr(parent, attr, value)
```

I like this a bit more with one level less of indentation. Semantically no change.